src/usr/local/www/system_authservers.php

```diff
@@ -132,11 +137,11 @@
 		}
 	}
 
-	/* Remove server from temp list used later on this page. */
-	unset($a_server[$_POST['id']]);
-
 	$savemsg = sprintf(gettext("Authentication Server %s deleted."), htmlspecialchars($serverdeleted));
 	write_config($savemsg);
+
+	/* Recreate the locally-used list to close up the deleted entry. */
+	$a_server = gen_auth_server_array();
```

Comment:
Using `$a_server = array_values($a_server);` here would fix the key IDs, or you could use the same code as the previous comment to recreate it.